jetty-websocket/websocket-jetty-common/src/main/java/org/eclipse/jetty/websocket/common/JettyWebSocketRemoteEndpoint.java

```diff
@@ -74,13 +75,18 @@ public void close()
      */
     public void close(int statusCode, String reason)
     {
-        try (SharedBlockingCallback.Blocker b = blocker.acquire())
+        if (!closed.compareAndSet(false, true))
```

Comment:
@gregw using this `AtomicBoolean` to make sure any subsequent closes are just a NOOP.

Reply:
But I think this should be moved to the CoreSession.